RHEL6 failure in TestE2E/SEQ/CONFIG/config_global/AllowSetuid

### Version of Singularity:

What version of Singularity are you using? Run:

```
3.5.2
```

### Actual behavior

```
--- FAIL: TestE2E/SEQ/CONFIG/config_global/AllowSetuid (0.11s)
                    singularitycmd.go:653: Running command "/usr/local/bin/singularity --debug config global --set allow setuid no"
                    config.go:373: Running command "/usr/local/bin/singularity --debug exec /tmp/stest.315551610/test.sif true"
                    config.go:373: 
                        "/usr/local/bin/singularity --debug exec /tmp/stest.315551610/test.sif true" output:
                        ESC[0mDEBUG   ESC[0m[U=500,P=30311]    persistentPreRunE()           Singularity version: 3.5.2+163-g6774e5e
                        ESC[0mDEBUG   ESC[0m[U=500,P=30311]    handleConfDir()               /home/dtrudg/.singularity already exists. Not creating.
                        ESC[0mDEBUG   ESC[0m[U=500,P=30311]    initCacheDir()                Creating cache directory: /tmp/stest.315551610/e2e-imgcache-139835724/cache
                        ESC[0mDEBUG   ESC[0m[U=500,P=30311]    initCacheDir()                Creating cache directory: /tmp/stest.315551610/e2e-imgcache-139835724/cache/library
                        ESC[0mDEBUG   ESC[0m[U=500,P=30311]    updateCacheSubdir()           Caching directory set to /tmp/stest.315551610/e2e-imgcache-139835724/cache/library
                        ESC[0mDEBUG   ESC[0m[U=500,P=30311]    initCacheDir()                Creating cache directory: /tmp/stest.315551610/e2e-imgcache-139835724/cache/oci-tmp
                        ESC[0mDEBUG   ESC[0m[U=500,P=30311]    updateCacheSubdir()           Caching directory set to /tmp/stest.315551610/e2e-imgcache-139835724/cache/oci-tmp
                        ESC[0mDEBUG   ESC[0m[U=500,P=30311]    initCacheDir()                Creating cache directory: /tmp/stest.315551610/e2e-imgcache-139835724/cache/oci
                        ESC[0mDEBUG   ESC[0m[U=500,P=30311]    updateCacheSubdir()           Caching directory set to /tmp/stest.315551610/e2e-imgcache-139835724/cache/oci
                        ESC[0mDEBUG   ESC[0m[U=500,P=30311]    initCacheDir()                Creating cache directory: /tmp/stest.315551610/e2e-imgcache-139835724/cache/net
                        ESC[0mDEBUG   ESC[0m[U=500,P=30311]    updateCacheSubdir()           Caching directory set to /tmp/stest.315551610/e2e-imgcache-139835724/cache/net
                        ESC[0mDEBUG   ESC[0m[U=500,P=30311]    initCacheDir()                Creating cache directory: /tmp/stest.315551610/e2e-imgcache-139835724/cache/shub
                        ESC[0mDEBUG   ESC[0m[U=500,P=30311]    updateCacheSubdir()           Caching directory set to /tmp/stest.315551610/e2e-imgcache-139835724/cache/shub
                        ESC[0mDEBUG   ESC[0m[U=500,P=30311]    initCacheDir()                Creating cache directory: /tmp/stest.315551610/e2e-imgcache-139835724/cache/oras
                        ESC[0mDEBUG   ESC[0m[U=500,P=30311]    updateCacheSubdir()           Caching directory set to /tmp/stest.315551610/e2e-imgcache-139835724/cache/oras
                        ESC[0mVERBOSE ESC[0m[U=500,P=30311]    execStarter()                 'allow setuid' set to 'no' by configuration, fallback to user namespace
                        ESC[0mDEBUG   ESC[0m[U=500,P=30311]    execStarter()                 Checking for encrypted system partition
                        ESC[0mDEBUG   ESC[0m[U=500,P=30311]    Init()                        Image format detection
                        ESC[0mDEBUG   ESC[0m[U=500,P=30311]    Init()                        Check for sandbox image format
                        ESC[0mDEBUG   ESC[0m[U=500,P=30311]    Init()                        sandbox format initializer returned: not a directory image
                        ESC[0mDEBUG   ESC[0m[U=500,P=30311]    Init()                        Check for sif image format
                        ESC[0mDEBUG   ESC[0m[U=500,P=30311]    Init()                        sif image format detected
                        ESC[0mVERBOSE ESC[0m[U=500,P=30311]    SetContainerEnv()             Not forwarding SINGULARITY_E2E_COVERAGE from user to container environment
                        ESC[0mVERBOSE ESC[0m[U=500,P=30311]    SetContainerEnv()             Not forwarding SINGULARITY_CACHEDIR from user to container environment
                        ESC[0mVERBOSE ESC[0m[U=500,P=30311]    SetContainerEnv()             Not forwarding SINGULARITY_SYPGPDIR from user to container environment
                        ESC[0mVERBOSE ESC[0m[U=500,P=30311]    SetContainerEnv()             HOME=/home/dtrudg
                        ESC[0mVERBOSE ESC[0m[U=500,P=30311]    execStarter()                 User namespace requested, convert image /tmp/stest.315551610/test.sif to sandbox
                        ESC[34mINFO    ESC[0m[U=500,P=30311]    execStarter()                 Convert SIF file to sandbox...
SC[0mDEBUG   ESC[0m[U=500,P=30311]    Init()                        Image format detection
                        ESC[0mDEBUG   ESC[0m[U=500,P=30311]    Init()                        Check for sandbox image format
                        ESC[0mDEBUG   ESC[0m[U=500,P=30311]    Init()                        sandbox format initializer returned: not a directory image
                        ESC[0mDEBUG   ESC[0m[U=500,P=30311]    Init()                        Check for sif image format
                        ESC[0mDEBUG   ESC[0m[U=500,P=30311]    Init()                        sif image format detected
                        ESC[0mDEBUG   ESC[0m[U=500,P=30311]    init()                        Use starter binary /usr/local/libexec/singularity/bin/starter
                        VERBOSE [U=500,P=30311]    print()                       Set messagelevel to: 5
                        ESC[0mVERBOSE [U=500,P=30311]    init()                        Starter initialization
                        ESC[0mDEBUG   [U=500,P=30311]    get_pipe_exec_fd()            PIPE_EXEC_FD value: 7
                        ESC[0mVERBOSE [U=500,P=30311]    is_suid()                     Check if we are running as setuid
                        ESC[0mDEBUG   [U=500,P=30311]    init()                        Read engine configuration
                        ESC[0mDEBUG   [U=500,P=30311]    init()                        Wait completion of stage1
                        ESC[0mDEBUG   [U=500,P=30330]    set_parent_death_signal()     Set parent death signal to 9
                        ESC[0mVERBOSE [U=500,P=30330]    init()                        Spawn stage 1
                        ESC[0mESC[0mDEBUG   ESC[0m[U=500,P=30330]    startup()                     singularity runtime engine selected
                        ESC[0mVERBOSE ESC[0m[U=500,P=30330]    startup()                     Execute stage 1
                        ESC[0mDEBUG   ESC[0m[U=500,P=30330]    StageOne()                    Entering stage 1
                        ESC[0mDEBUG   ESC[0m[U=500,P=30330]    prepareAutofs()               Found "/misc" as autofs mount point
                        ESC[0mDEBUG   ESC[0m[U=500,P=30330]    prepareAutofs()               Found "/net" as autofs mount point
                        ESC[0mDEBUG   ESC[0m[U=500,P=30330]    prepareAutofs()               Could not keep file descriptor for bind path /etc/localtime: no mount point
                        ESC[0mDEBUG   ESC[0m[U=500,P=30330]    prepareAutofs()               Could not keep file descriptor for bind path /etc/hosts: no mount point
                        ESC[0mDEBUG   ESC[0m[U=500,P=30330]    prepareAutofs()               Could not keep file descriptor for home directory /home/dtrudg: no mount point
                        ESC[0mDEBUG   ESC[0m[U=500,P=30330]    prepareAutofs()               Could not keep file descriptor for current working directory /home/dtrudg/singularity/e2e: no mount point
                        ESC[0mDEBUG   ESC[0m[U=500,P=30330]    Init()                        Image format detection
                        ESC[0mDEBUG   ESC[0m[U=500,P=30330]    Init()                        Check for sandbox image format
                        ESC[0mDEBUG   ESC[0m[U=500,P=30330]    Init()                        sandbox image format detected
                        ESC[0mDEBUG   ESC[0m[U=500,P=30330]    setSessionLayer()             Using underlay layer: user namespace requested
                        VERBOSE [U=500,P=30311]    wait_child()                  stage 1 exited with status 0
                        ESC[0mDEBUG   [U=500,P=30311]    init()                        Applying stage 1 working directory
                        ESC[0mDEBUG   [U=500,P=30311]    cleanup_fd()                  Close file descriptor 4
                        ESC[0mDEBUG   [U=500,P=30311]    init()                        Set child signal mask
                        ESC[0mDEBUG   [U=500,P=30311]    init()                        Create socketpair for master communication channel
                        ESC[0mDEBUG   [U=500,P=30311]    init()                        Create RPC socketpair for communication between stage 2 and RPC server
                        ESC[0mVERBOSE [U=500,P=30311]    user_namespace_init()         Create user namespace
                        ESC[0mVERBOSE [U=500,P=30311]    create_namespace()            Create user namespace
                        ESC[0mESC[91mERROR   [U=500,P=30311]    init()                        Failed to create user namespace: Invalid argument
                        ESC[0m
                    config.go:373: got 1 as exit code and was expecting 0: exit status 1
                        waiting for command "/usr/local/bin/singularity --debug exec /tmp/stest.315551610/test.sif true"
```


### What OS/distro are you running

```
RHEL 6
```

--- a/e2e/config/config.go
+++ b/e2e/config/config.go
@@ -10,6 +10,7 @@ import (
 
 	"github.com/sylabs/singularity/e2e/internal/e2e"
 	"github.com/sylabs/singularity/e2e/internal/testhelper"
+	"github.com/sylabs/singularity/internal/pkg/test/tool/require"
 	"github.com/sylabs/singularity/internal/pkg/util/user"
 )
 
@@ -46,22 +47,27 @@ func (c configTests) configGlobal(t *testing.T) {
 	}
 
 	tests := []struct {
-		name           string
-		argv           []string
-		profile        e2e.Profile
-		cwd            string
-		directive      string
-		directiveValue string
-		exit           int
-		resultOp       e2e.SingularityCmdResultOp
+		name              string
+		argv              []string
+		profile           e2e.Profile
+		addRequirementsFn func(*testing.T)
+		cwd               string
+		directive         string
+		directiveValue    string
+		exit              int
+		resultOp          e2e.SingularityCmdResultOp
 	}{
 		{
-			name:           "AllowSetuid",
-			argv:           []string{c.env.ImagePath, "true"},
-			profile:        e2e.UserProfile,
-			directive:      "allow setuid",
-			directiveValue: "no",
-			exit:           0,
+			name: "AllowSetuid",
+			argv: []string{c.env.ImagePath, "true"},
+			// We are testing if we fall back to user namespace without `--userns`
+			// so we need to use the UserProfile, and check separately if userns
+			// support is possible.
+			profile:           e2e.UserProfile,
+			addRequirementsFn: require.UserNamespace,
+			directive:         "allow setuid",
+			directiveValue:    "no",
+			exit:              0,
 		},
 		{
 			name:           "MaxLoopDevices",
@@ -376,6 +382,9 @@ func (c configTests) configGlobal(t *testing.T) {
 			e2e.WithProfile(tt.profile),
 			e2e.WithDir(tt.cwd),
 			e2e.PreRun(func(t *testing.T) {
+				if tt.addRequirementsFn != nil {
+					tt.addRequirementsFn(t)
+				}
 				setDirective(t, tt.directive, tt.directiveValue)
 			}),
 			e2e.PostRun(func(t *testing.T) {

